[Bug]: LuisaRender build failure with newer CMake (PyTorch 2.8+ Docker image)

### Bug Description

I design my own docker image according to [Dockerfile](https://github.com/Genesis-Embodied-AI/Genesis/blob/main/docker/Dockerfile). It works fine until https://github.com/Genesis-Embodied-AI/Genesis/pull/2034 changes the minimum torch version.

I switch my base image from `pytorch/pytorch:2.6.0-cuda12.4-cudnn9-devel` to `pytorch/pytorch:2.8.0-cuda12.8-cudnn9-devel` and the following error occurs during LuisaRender building. The full output is also provided.

```bash
5.900 CMake Error at src/compute/src/ext/EASTL/packages/EABase/CMakeLists.txt:4 (cmake_minimum_required):
5.900   Compatibility with CMake < 3.5 has been removed from CMake.
5.900 
5.900   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
5.900   to tell CMake that the project requires at least <min> but has been updated
5.900   to work with policies introduced by <max> or earlier.
5.900 
5.900   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
5.900 
5.900 
5.902 -- Configuring incomplete, errors occurred!
```

I resolve this problem by adding `-D CMAKE_POLICY_VERSION_MINIMUM=3.5` into `build_luisa.sh`. It seems that both Dockerfile and build_luisa.sh should be updated after bumping up torch version.

### Steps to Reproduce

```
RUN chmod +x ./build_luisa.sh && \
    ./build_luisa.sh ${PYTHON_VERSION}
```

### Expected Behavior

build complete.

### Screenshots/Videos

_No response_

### Relevant log output

```shell
[24/30] RUN chmod +x ./build_luisa.sh && ./build_luisa.sh 3.11:                                                           
4.714 -- The C compiler identification is GNU 11.4.0                                                                         
4.846 -- The CXX compiler identification is GNU 11.4.0                                                                       
4.857 -- Detecting C compiler ABI info                                                                                       
4.916 -- Detecting C compiler ABI info - done                                                                                
4.933 -- Check for working C compiler: /usr/bin/cc - skipped                                                                 
4.934 -- Detecting C compile features                                                                                        
4.934 -- Detecting C compile features - done                                                                                 
4.982 -- Detecting CXX compiler ABI info                                                                                     
5.104 -- Detecting CXX compiler ABI info - done                                                                              
5.121 -- Check for working CXX compiler: /usr/bin/c++ - skipped                                                              
5.121 -- Detecting CXX compile features                                                                                      5.122 -- Detecting CXX compile features - done                                                                               
5.255 -- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.11")                                              
5.257 -- Shared libraries enabled
5.266 -- Looking for ZLIB...
5.270 -- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2")
5.271 -- Checking for module 'zzip-zlib-config'
5.273 --   No package 'zzip-zlib-config' found
5.285 -- Checking for module 'minizip'
5.287 --   No package 'minizip' found           
5.287 -- Build an import-only version of Assimp.
5.294 -- Enabled importer formats: AMF 3DS AC ASE ASSBIN B3D BVH COLLADA DXF CSM HMP IRRMESH IQM IRR LWO LWS M3D MD2 MD3 MD5 
MDC MDL NFF NDO OFF OBJ OGRE OPENGEX PLY MS3D COB BLEND IFC XGL FBX Q3D Q3BSP RAW SIB SMD STL TERRAGEN 3D X X3D GLTF 3MF MMD
5.294 -- Disabled importer formats:                                                                                          
5.302 -- Build for Linux-5.15.0-139-generic (x86_64)                                                                         
5.302 CMake Warning at src/compute/scripts/validate_options.cmake:70 (message):                                             
5.302   The DirectX backend is not available.  The DirectX backend will be                                                   
5.302   disabled.                                                                                                            
5.302 Call Stack (most recent call first):                                                                                   
5.302   src/compute/scripts/validate_options.cmake:80 (report_feature_not_available)                                         
5.302   src/compute/CMakeLists.txt:57 (include)                                                                              
5.302                                                                                                                        
5.302                                                                                                                        
5.302 CMake Warning at src/compute/scripts/validate_options.cmake:70 (message):                                              
5.302   The Metal backend is not available.  The Metal backend will be disabled.                                             
5.302 Call Stack (most recent call first):                                                                                   
5.302   src/compute/scripts/validate_options.cmake:91 (report_feature_not_available)                                         
5.302   src/compute/CMakeLists.txt:57 (include)                                                                              
5.302                                                                                                                        
5.302                                                                                                                        
5.318 -- Performing Test CMAKE_HAVE_LIBC_PTHREAD                                                                             
5.437 -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success                                                                   
5.439 -- Found Threads: TRUE                                                                                                 
5.573 -- Could NOT find LLVM (missing: LLVM_DIR)                                                                             
5.585 -- Could NOT find embree (missing: embree_DIR)                                                                         
5.585 CMake Warning at src/compute/scripts/validate_options.cmake:70 (message):                                              
5.585   The fallback backend is not available.  The fallback backend will be                                     
5.585   disabled.                                                                                                            
5.585 Call Stack (most recent call first): 
5.585   src/compute/scripts/validate_options.cmake:116 (report_feature_not_available)                               [114/373]
5.585   src/compute/CMakeLists.txt:57 (include)                                                                              
5.585                                                                                                                        
5.585                                                                                                                        
5.589 -- Could not find sccache                                                                                              
5.609 -- Build spdlog: 1.15.1                                                                                                
5.615 -- Build type: Release                                                                                                 
5.616 -- Looking for fwrite_unlocked                                                                                         
5.717 -- Looking for fwrite_unlocked - not found                                                                             
5.722 -- Performing Test EASTL_HAS_FCHAR8T_FLAG                                                                              
5.834 -- Performing Test EASTL_HAS_FCHAR8T_FLAG - Success                                                                    
5.835 -- Performing Test EASTL_HAS_ZCCHAR8T_FLAG                                                                             
5.899 -- Performing Test EASTL_HAS_ZCCHAR8T_FLAG - Failed                                                                    
5.900 CMake Error at src/compute/src/ext/EASTL/packages/EABase/CMakeLists.txt:4 (cmake_minimum_required):                   
5.900   Compatibility with CMake < 3.5 has been removed from CMake.                                                         
5.900                                                                                                                        
5.900   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax                                          
5.900   to tell CMake that the project requires at least <min> but has been updated                                         
5.900   to work with policies introduced by <max> or earlier.                                                                
5.900                                                                                                                        
5.900   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.                                               
5.900                                                                                                                        
5.900                                                                                                                        
5.902 -- Configuring incomplete, errors occurred!                                                                            
------                                                                                                                       
Dockerfile:285                                                                                                               
--------------------                                                                                                         
 283 |     # COPY build_luisa.sh /workspace/build_luisa.sh                                                                   
 284 |     RUN cp /workspace/Genesis/docker/build_luisa.sh /workspace/build_luisa.sh                                        
 285 | >>> RUN chmod +x ./build_luisa.sh && ./build_luisa.sh ${PYTHON_VERSION}                                              
 286 |                                                                                                                       
 287 |     # install Genesis From Source                                                                                     
--------------------                                                                                                         
ERROR: failed to solve: process "/bin/sh -c chmod +x ./build_luisa.sh && ./build_luisa.sh ${PYTHON_VERSION}" did not complete
 successfully: exit code: 1
```

### Environment

Docker image based on `pytorch/pytorch:2.8.0-cuda12.8-cudnn9-devel`


### Release version or Commit ID

a2787f5fa1aeac250c4f36702b5de949993ec057

### Additional Context

_No response_

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -1,12 +1,12 @@
-ARG CUDA_VERSION=12.1
+ARG CUDA_VERSION=12.8
 
 # ===============================================================
 # Stage 1: Build LuisaRender
 # ===============================================================
-FROM pytorch/pytorch:2.5.1-cuda${CUDA_VERSION}-cudnn9-devel AS builder
+FROM pytorch/pytorch:2.11.0-cuda${CUDA_VERSION}-cudnn9-devel AS builder
 
 ENV DEBIAN_FRONTEND=noninteractive
-ARG PYTHON_VERSION=3.11
+ARG PYTHON_VERSION=3.12
 
 # Install necessary packages
 RUN apt-get update && apt-get install -y --no-install-recommends \
@@ -38,10 +38,7 @@ RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
 RUN pip install "pybind11[global]"
 
 # Install CMake
-RUN wget https://github.com/Kitware/CMake/releases/download/v3.31.0-rc2/cmake-3.31.0-rc2-linux-x86_64.sh && \
-    chmod +x cmake-3.31.0-rc2-linux-x86_64.sh && \
-    ./cmake-3.31.0-rc2-linux-x86_64.sh --skip-license --prefix=/usr/local && \
-    rm cmake-3.31.0-rc2-linux-x86_64.sh
+RUN pip install --no-cache-dir cmake==3.31.6
 
 # Build LuisaRender
 WORKDIR /workspace
@@ -54,9 +51,10 @@ RUN chmod +x ./build_luisa.sh && ./build_luisa.sh ${PYTHON_VERSION}
 # ===============================================================
 # Stage 2: Runtime Environment
 # ===============================================================
-FROM pytorch/pytorch:2.5.1-cuda${CUDA_VERSION}-cudnn9-devel
+FROM pytorch/pytorch:2.11.0-cuda${CUDA_VERSION}-cudnn9-devel
 
-ARG PYTHON_VERSION=3.11
+ARG PYTHON_VERSION=3.12
+ENV PYTHON_VERSION=${PYTHON_VERSION}
 ENV DEBIAN_FRONTEND=noninteractive
 ENV NVIDIA_DRIVER_CAPABILITIES=all
 
@@ -98,12 +96,12 @@ RUN git clone https://github.com/Genesis-Embodied-AI/Genesis.git && \
 
 # -------------------- Surface Reconstruction --------------------
 # Set the LD_LIBRARY_PATH directly in the environment
-COPY --from=builder /workspace/Genesis/genesis/ext/ParticleMesher/ParticleMesherPy /opt/conda/lib/python3.11/site-packages/genesis/ext/ParticleMesher/ParticleMesherPy
-ENV LD_LIBRARY_PATH=/opt/conda/lib/python3.11/site-packages/genesis/ext/ParticleMesher/ParticleMesherPy:$LD_LIBRARY_PATH
+COPY --from=builder /workspace/Genesis/genesis/ext/ParticleMesher/ParticleMesherPy /opt/conda/lib/python${PYTHON_VERSION}/site-packages/genesis/ext/ParticleMesher/ParticleMesherPy
+ENV LD_LIBRARY_PATH=/opt/conda/lib/python${PYTHON_VERSION}/site-packages/genesis/ext/ParticleMesher/ParticleMesherPy:$LD_LIBRARY_PATH
 
 # --------------------- Ray Tracing Renderer ---------------------
 # Copy LuisaRender build artifacts from the builder stage
-COPY --from=builder /workspace/Genesis/genesis/ext/LuisaRender/build/bin /opt/conda/lib/python3.11/site-packages/genesis/ext/LuisaRender/build/bin
+COPY --from=builder /workspace/Genesis/genesis/ext/LuisaRender/build/bin /opt/conda/lib/python${PYTHON_VERSION}/site-packages/genesis/ext/LuisaRender/build/bin
 # fix GLIBCXX_3.4.30 not found
 # Remove this — it's dangerous and broke cmake
 # RUN cd /opt/conda/lib && \
@@ -120,7 +118,7 @@ RUN echo '#!/bin/bash\n\
 if [ -n "${LOCAL_USER_ID}" ]; then\n\
   echo "Starting with UID: $LOCAL_USER_ID"\n\
   usermod -u $LOCAL_USER_ID user\n\
-  chown -R user:user /opt/conda/lib/python3.11/site-packages/genesis/ext/LuisaRender/\n\
+  chown -R user:user /opt/conda/lib/python${PYTHON_VERSION}/site-packages/genesis/ext/LuisaRender/\n\
 fi\n\
 exec /bin/bash' > /entrypoint.sh && \
 chmod +x /entrypoint.sh

--- a/docker/build_luisa.sh
+++ b/docker/build_luisa.sh
@@ -13,6 +13,7 @@ git submodule update --init --recursive && \
 mkdir -p build && \
 cmake -S . -B build \
     -D CMAKE_BUILD_TYPE=Release \
+    -D CMAKE_POLICY_VERSION_MINIMUM=3.5 \
     -D PYTHON_VERSIONS=$PYTHON_VERSION \
     -D LUISA_COMPUTE_DOWNLOAD_NVCOMP=ON \
     -D LUISA_COMPUTE_DOWNLOAD_OIDN=ON \

